packages/cli/src/active-executions.ts

```diff
@@ -111,8 +113,10 @@ export class ActiveExecutions {
 			})
 			.finally(() => {
 				this.concurrencyControl.release({ mode: executionData.executionMode });
-				delete this.activeExecutions[executionId];
-				this.logger.debug('Execution removed', { executionId });
+				if (this.activeExecutions[executionId]?.status !== 'waiting') {
```

Comment:
Waiting executions will now take up a concurrency slot, even though technically they're not really running.

I cannot say if this should or should not be case, but bringing this up to make sure it's a conscious decision.

Reply:
on line 115, we are still releasing the concurrency slot back.

Reply:
I'm blind today 🙈